superset/examples/utils.py

```diff
@@ -44,28 +44,30 @@ def load_examples_from_configs(
 
 def load_contents(load_test_data: bool = False) -> dict[str, Any]:
     """Traverse configs directory and load contents"""
-    root = Path("examples/configs")
-    resource_names = resource_listdir("superset", str(root))
-    queue = [root / resource_name for resource_name in resource_names]
+    root = files("superset") / "examples/configs"
+    resource_names = (files("superset") / str(root)).iterdir()
+    queue = [root / str(resource_name) for resource_name in resource_names]
 
     contents: dict[Path, str] = {}
     while queue:
         path_name = queue.pop()
         test_re = re.compile(r"\.test\.|metadata\.yaml$")
 
-        if resource_isdir("superset", str(path_name)):
+        if (files("superset") / str(path_name)).is_dir():
             queue.extend(
-                path_name / child_name
-                for child_name in resource_listdir("superset", str(path_name))
+                path_name / str(child_name)
```

Comment:
Why is `str(...)` needed here?

Reply:
Good question. I'll double check. I might have been blindly following `mypy` errors. Now I'm not sure if the `mypy` errors were even valid errors.

Reply:
Ok. I did some digging on `joinpath()`.

`mypy` was correct. The argument to `joinpath()` is defined to be of `StrPath` type which is defined as `Union[str, PathLike[str]]`

So, whilst `Traversable.join_path(Traversable)` will execute in CPython (likely due to coercion), it breaks the API protocol of `joinpath()`